docs/source/distribution_implementation.md

```diff
@@ -0,0 +1,425 @@
+# Implementing a Distribution (developer guide)
+
+This guide provides an overview on how to implement a distribution for version 4 of PyMC3.
+It is designed for developers who wish to add a new distribution to the library.
+Users will not be aware of all this complexity and should instead make use of helper methods such as (TODO).
```

Comment:
Leftover TODO

Reply:
It's still a pymc3 to do. This was the role of `Densitydist` but that was deprecated or at least not refactored for V4. Maybe we can link something like: https://github.com/pymc-devs/pymc-examples/pull/185